server/routes/api/import_detection/detection.py

```diff
@@ -194,3 +202,148 @@ def detect_columns(col_order: List[Column],
+        column_values: Dict[int, List[str]]. This is the mapping from column indices (int) to column values (List[str]).
+    
+        All values (indices) in 'column_idx_order' are expected to be present as keys all the other maps. If a key is not
+        found, an error is returned. If any parameter is not of the expected format, an error is returned.
+
+    Returns:
+        A json response of the detected columns. Sample json response:
+
+        The json string is of the following form (examples):
+
+        Example # 1: Date is detected in one column.
+        [
+            Place: {
+                type : column,
+                column: None,
+                place_type: {
+                    dcid: Country,
+                    display_name: Country,
+                }
+                place_property: {
+                    dcid: isoCode,
+                    display_name: ISO Code,
+                },
+                headers: None,
+            },
+            Date: {
+                type : column,
+                place_type: None
+                place_property: None,
+                headers: None,
+            }
+        ]
+
+        Example # 2: Date is detected in column headers.
+        [
+            Place: {
+                type : column,
+                column: None,
+                place_type: {
+                    dcid: Country,
+                    display_name: Country,
+                }
+                place_property: {
+                    dcid: isoCode,
+                    display_name: ISO Code,
+                },
+                headers: None,
+            },
+            Date: {
+                type : column,
+                place_type: None,
+                place_property: None,
+                headers: [
+                    Column{
+                        id: name_0,
+                        header: name,
+                        column_idx: 0,
+                    },
+                    Column{
+                        id: name_name_1,
+                        header: name_name,
+                        column_idx: 1,
+                    },
+                    ...
+                ],
+            }
+        ]
+    """
+    # Parse and validate the query params.
+    column_order: List[Column] = []
+    js = json.loads(request.get_json())
+    try:
+        column_idx_order: List[int] = [int(v) for v in js["column_idx_order"]]
+        column_ids: Dict[int, str] = {
+            int(k): v for k, v in js["column_ids"].items()
+        }
+        column_headers: Dict[int, str] = {
+            int(k): v for k, v in js["column_headers"].items()
+        }
+        column_values: Dict[int, List[str]] = {
+            int(k): v for k, v in js["column_values"].items()
+        }
+
+        if not column_idx_order or not column_ids or not column_headers or not column_values:
+            raise KeyError()
```

Comment:
we should raise KeyError for each key individually so we can include which key is missing.

Reply:
No longer necessary since we shouldn't ever get to this because the missing key error should be hit earlier anyway.